src/foam/box/HTTPBox.js

```diff
@@ -26,8 +26,9 @@ foam.CLASS({
   ],
 
   imports: [
+    'generator? as ctxGenerator',
```

Comment:
> I would have to subclass all Box/Service classes to do that.

No you would only have to subclass the two you want to replace (MessagePortService and MessagePortBox) and then use the context to replace which ones get created throughout the rest of the box system.

> All I want to do is customize serialization,

I know it may seem like you want a simple thing, but every knob we add the framework adds complexity and overhead for all other organizations that want to use it.  It's much better if the framework only has one canonical way to solve particular problems.  If every organization added knobs and customization hooks for their use cases it would be unmaintainable.

> is correct (and fast).

If it's both correct and fast then we could consider moving to it.  However your serialization approach right now creates unnecessary garabage by doing objectify and then JSON.stringify.  If you turn strict:true on in the existing serializers it is supposed to output proper json that you could use JSON.parse on, and this wouldn't require the intermediate objectify() step, which will hoepfully create less garbage.

> [1] foam.json's parse/stringify is "effectively" parse(parseString(str)) / stringify(objectify(o))

stringify doesn't go through an intermediate objectify step, it goes straight from the source object to the string.  objectify() was added for browser APIs that need plain old javascript objects rather than strings of JSON.